### PR TITLE
refactor!: Introduce regular surface intermediate class

### PR DESCRIPTION
**Pull request by paulgessinger**
This adds a new class

```cpp
class RegularSurface : public Surface {};
```
that reimplements a number of functions while discarding the `direction` argument.

### CODE REVIEW

**Comment by paulgessinger** [comment]
I extracted the changes to `normal` and the introduction of `coerceToSurface` to #2344, so we can evaluate whether this is a good idea separately.

File: Tests/UnitTests/Core/Vertexing/IterativeVertexFinderTests.cpp [hunk 1]
```diff
@@ -264,10 +264,8 @@ BOOST_AUTO_TEST_CASE(iterative_finder_test) {
     // find vertices
     auto res = finder.find(tracksPtr, vertexingOptions, state);
 
-    BOOST_CHECK(res.ok());
-
     if (!res.ok()) {
-      std::cout << res.error().message() << std::endl;
+      BOOST_FAIL(res.error().message());
```
Comment (asalzburger) [inline Tests/UnitTests/Core/Vertexing/IterativeVertexFinderTests.cpp:268]: Good catch.

**Review by asalzburger — DISMISSED** [review]
I like this very much - with the consistent usage it becomes clear what a boundary/portal can be.

**Comment by asalzburger** [comment]
It's sort of breaking and not breaking at the same time?

**Comment by paulgessinger** [comment]
Yeah I guess it depends what we consider breaking, indeed.

File: Core/include/Acts/Surfaces/RegularSurface.hpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Utilities/ThrowAssert.hpp"
+
+namespace Acts {
+
+class RegularSurface : public Surface {
```
Comment (andiwand) [inline Core/include/Acts/Surfaces/RegularSurface.hpp:16]:
```suggestion
/// A physical surface which does not depend on the direction you look at it from. As such it narrows the interface of @class Surface and allows inspection without providing a global position and direction.
class RegularSurface : public Surface {
```

should we add a small into to this class?

File: Examples/Python/python/acts/examples/__init__.py [hunk 3]
```diff
@@ -358,7 +358,7 @@ def __init__(self, *args, **kwargs):
                     n.append(self.FpeMask(loc, t, count))
                 kwargs["fpeMasks"] = n
 
-            kwargs["fpeMasks"] = kwargs.get("fpeMasks", []) + self._getAutoFpeMasks()
+        kwargs["fpeMasks"] = kwargs.get("fpeMasks", []) + self._getAutoFpeMasks()
```
Comment (andiwand) [inline Examples/Python/python/acts/examples/__init__.py:361]: I guess this comes from a different branch?

File: Tests/UnitTests/Core/Surfaces/SurfaceTests.cpp [hunk 4]
```diff
@@ -111,12 +111,12 @@ BOOST_AUTO_TEST_CASE(SurfaceProperties) {
   RotationMatrix3 unitary;
   unitary << 1, 0, 0, 0, 1, 0, 0, 0, 1;
   auto referenceFrame = surface.referenceFrame(
-      tgContext, reference, mom);  // need more complex case to test
+      tgContext, Vector3{1, 2, 3}, mom);  // need more complex case to test
```
Comment (andiwand) [inline Tests/UnitTests/Core/Surfaces/SurfaceTests.cpp:114]: in the general case this vector should be normalized I think

File: Tests/UnitTests/Core/Surfaces/SurfaceTests.cpp [hunk 5]
```diff
@@ -111,12 +111,12 @@ BOOST_AUTO_TEST_CASE(SurfaceProperties) {
...
   BOOST_CHECK_EQUAL(referenceFrame, unitary);
   // normal()
-  auto normal = surface.Surface::normal(tgContext,
-                                        reference);  // needs more complex
-                                                     // test
+  auto normal = surface.normal(tgContext, Vector3{1, 2, 3},
```
Comment (andiwand) [inline Tests/UnitTests/Core/Surfaces/SurfaceTests.cpp:117]: same here

**Review by andiwand — COMMENTED** [review]
nice! left a few minor comments

File: Examples/Python/python/acts/examples/__init__.py (hunk 3 above)
Comment (paulgessinger) [reply Examples/Python/python/acts/examples/__init__.py:361]: Right, I've moved this here and there because I was having issues.

**Comment by paulgessinger** [comment]
Updated now. Let's err on the safe side though and hold this for `v30`.

**Review by andiwand — APPROVED** [review]

**Review by andiwand — APPROVED** [review]
:rocket:

**Comment by paulgessinger** [comment]
I don't have time to fix this right now. Let's push this again.

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [f1208831ce16b3031f54d79cb12dabf5620dd693]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33431389) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6388966)!

**Comment by paulgessinger** [comment]
This is rebased again. Should we just merge it @andiwand?

**Review by andiwand — APPROVED** [review]

**Review by andiwand — APPROVED** [review]

**Comment by paulgessinger** [comment]
Ok I might have to fix some compile issues and also the Athena build seems to be failing on this in an unexpected way.

**Comment by paulgessinger** [comment]
Ready again now I think @andiwand.

**Review by andiwand — APPROVED** [review]

**Pull request merged**